### ISSUE TITLE
How I can contiuned use update? 

### ISSUE THREAD

**Issue by ying2025** [artjomb/cryptojs-extension#6]
I want use it like this picture.

![image](https://user-images.githubusercontent.com/28092539/44964649-2ceb2b00-af64-11e8-86e2-25f75cd224c1.png)
The code is:

     function test3() {
         let vec = {
            key: "8395FCF1E95BEBD697BD010BC766AAC3",
            nonce: "22E7ADD93CFC6393C57EC0B3C17D6B44",
            header: "126735FCC320D25A",
            msg: "CA40D7446E545FFAED3BD12A740A659FFBBB3CEAB7",
            msg2: "6CF36720872B8513F6EAB1A8A44438D5EF11",
            ct: "CB8920F87A6C75CFF39627B56E3ED197C552D295A7CFC46AFC253B4652B1AF3795B124AB6E"
        };
        let keyBytes = CryptoJS.enc.Hex.parse(vec.key),
            msgBytes = CryptoJS.enc.Hex.parse(vec.msg),
            msgBytes2 = CryptoJS.enc.Hex.parse(vec.msg2),
            nonceBytes = CryptoJS.enc.Hex.parse(vec.nonce),
            headerBytes = CryptoJS.enc.Hex.parse(vec.header);   
        let eax = CryptoJS.EAX.create(keyBytes);
        eax.encrypt(nonceBytes, [headerBytes]);
        eax.update(msgBytes);
        eax.update(msgBytes2);
        let et = eax.finalize();
        // console.log(et.toString())
        console.log("msgB1 ",vec.msg.toLowerCase(), "ciphertext match ["+"]");

    }.

![image](https://user-images.githubusercontent.com/28092539/44964678-599f4280-af64-11e8-92b3-492104034e81.png)
![image](https://user-images.githubusercontent.com/28092539/44964717-910def00-af64-11e8-83e5-7b0321e664f1.png)

![image](https://user-images.githubusercontent.com/28092539/44964690-6de33f80-af64-11e8-8429-f6e8fb6a26fd.png)
![image](https://user-images.githubusercontent.com/28092539/44964699-78053e00-af64-11e8-924b-609f3610d472.png)

**Comment by artjomb** [comment]
@ying2025 I don't see how having `encrypt` and `update`+`finalize` makes sense semantically. `encrypt` is complete and closed operation, whereas `create`+`updateAAD`+`initCrypt`+`update`+`finalize` is the fully self-managed way to do encrypt. If you want to combine `updateAAD`+`initCrypt`, why not call it `prepareEncryption` or even combine `create`+`updateAAD`+`initCrypt` and overwrite `create` or call it `createAndInit`? The latter would need to call the inherited function from the underlying framework somehow.

**Comment by ying2025** [comment]
Yes, you are right. I have modified it. But now  I continuously use update, the message can decrypt correctly, but ext.equals(self._tag, self._buf) == false.
If it use multiple update, I put the encrypt data to the self.xoredData avoiding being covered. I don't konw whether it right, and how to deal the last block.
![image](https://user-images.githubusercontent.com/28092539/45017877-4f13a480-b05b-11e8-9d00-b93c4c19dd25.png)

![image](https://user-images.githubusercontent.com/28092539/45018089-d6611800-b05b-11e8-847a-58940dad186a.png)


And M_i is diffierent when update in encrtpt and update in decrypt. 
![image](https://user-images.githubusercontent.com/28092539/45017635-a8c79f00-b05a-11e8-983c-cfd7642e4297.png)

**Comment by ying2025** [comment]
I change the postion of self._mac.update for message, and it can decrypt rightly, but I don't konw whether it have any problems.
![image](https://user-images.githubusercontent.com/28092539/45133489-09c0b580-b1c8-11e8-881a-9f811f33b324.png)
